src/docs/arch-transaction.dox

```diff
@@ -5,4 +5,114 @@ A caller of WiredTiger uses @ref transactions within the API to start and stop t
 a session (thread of control).
 
 Internally, the current transaction state is represented by the WT_TXN structure.
+
+WiredTiger performs all the operations in transactions. If the user doesn't explicitly declare a transaction, WiredTiger will automatically create a transaction for the user's operation.
+
+@section Lifecycle
+
+A WiredTiger session creates and managers the transactions' whole lifecycle. A session is only able to manage one transaction at any given time and it cannot run another transaction until the current one completes. Since every session are executed on a single thread, all the operations in the transaction are executed sequentially on the same thread as well.
```

Comment:
The word whole here isn't necessar.

Reply:
Another comment for this line, I think the wording is a bit odd, I would phrase this as "One one transaction can be run at a time per session, and that transaction must complete before another transaction can be started." or something. Additionally "Since every session are executed on a single thread" seems grammatically weird. "Since every session is singly-threaded all the operations in the transaction are executed on the same thread".